enhancements/operator-conditions.md

```diff
@@ -0,0 +1,180 @@
+
+| Field | Description |
+| ----- | ----------- |
+| Type | Type of condition in CamelCase or in foo.example.com/CamelCase. |
+| Status | Status of the condition, one of True, False, Unknown. |
+| ObservedGeneration | If set, this represents the .metadata.generation that the condition was set based upon. |
+| LastTransitionTime | Last time the condition transitioned from one status to another. |
+| Reason | The reason for the condition's last transition in CamelCase. |
+| Message | A human readable message indicating details about the transition. |
+
+OLM will define a set of "OLM Supported Conditions" which will influence how OLM manages the Lifecycle of the operator when present.
+
+### OLM Supported Conditions
+
+All "OLM Supported Conditions" will be communicated by setting the conditions in the `status.conditions` array of the `OperatorHealth` CR associated with the operator. The name of each "OLM Supported Condition" will correspond to the `Type` field of the condition added to the array. Let's look at a fake "OLM Supported Condition" named `foo` that has been added to an `OperatorHealth` CR:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: OperatorHealth
+metadata:
+  name: foo-operator.v1.0.0
+  namespace: operators
+status:
+   conditions:
+    type: foo # The name of the `foo` OLM Supported Condition.
+    status: "True" # The "status" of the `foo` OLM Supported Condition.
+    reason: "foo" # A camelCase reason that the operator is in this state
+    message: "The operator is in the "foo" state." # A human readable message.
+    lastTransitionTime: "2020-08-24T23:15:55Z" # The last time the condition transitioned from one status to another.
+```
+
+#### Upgradeable
+
+The Upgradeable "OLM Supported Condition" allows an operator to communicate when OLM should or should not upgrade the operator.
+
+When the Upgradeable condition is set to `False`, OLM will not:
+
+- Perform automatic or manual updates to the operator.
+- Update the operator as a part of [dependency resolution](./operator-dependency-resolution.md).
+
+When the Upgradeable condition is set to `True`, OLM will not prevent upgrades to the operator.
+
+The Upgradeable condition might be useful when:
+
+- An operator is about to start a critical process and should not be upgraded until after the process is completed.
+- The operator is performing a migration of CRs that must be completed before the operator is ready to be upgraded.
+
+##### Example OperatorHealth with the Upgradeable Condition
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: OperatorHealth
+metadata:
+  name: foo-operator.v1.0.0
+  namespace: operators
+status:
+   conditions:
+    type: upgradeable # The name of the `foo` OLM Supported Condition.
```

Comment:
should this be uppercase? I believe that's the convention for kube enums.

Reply:
I checked existing conditions on the pod resources and it looks like Types are PascalCase:
```
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2020-08-25T20:01:30Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2020-08-25T20:01:34Z"
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2020-08-25T20:01:34Z"
    status: "True"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2020-08-25T20:01:30Z"
    status: "True"
    type: PodScheduled
```